### PR TITLE
Make Result.Interface() return an empty string, as is.

### PR DESCRIPTION
**Pull request by ozevren**
**What this PR does / why we need it**:
+ Removing the old handling code that converts an empty string to
nil.
+ Bolster stringmap lookup & or tests (i.e. ar["c"] | "foo").

With this, the semantics for expression: `ar["c"] | "foo"` are:

bag {} => "foo"
bag {"ar": {} } => "foo"
bag {"ar": { "c":"b" } => "b"

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
https://github.com/istio/old_mixer_repo/issues/934

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: mixer/pkg/il/compiler/compiler_test.go [hunk 1]
```diff
@@ -792,6 +806,13 @@ end`,
 	{
 		expr: `ar["c"] | "foo"`,
 		input: map[string]interface{}{
+			"ar": map[string]string{},
+		},
+		result: "foo",
+	},
+	{
+		expr: `ar["c"] | "foo"`,
+		input: map[string]interface{}{
```
Comment (guptasu) [inline mixer/pkg/il/compiler/compiler_test.go]: Consider adding test for below cases where the attribute reference (source.absentInBag) does not exist . May be it already exists.

attribute reference | ""
attribute reference | "foo"

Comment (ozevren) [reply mixer/pkg/il/compiler/compiler_test.go]: There are tests for the second case already.
https://github.com/istio/istio/blob/master/mixer/pkg/il/compiler/compiler_test.go#L414
The first case simply returns "" as expected. I'll add tests in my follow-up test-refactoring pack.

**Pull request merged**